src/software/backend/input/network/networking/proto_multicast_listener.h

```diff
@@ -41,15 +42,19 @@ class SSLGameControllerClient
     void handleDataReception(const boost::system::error_code& error,
                              size_t num_bytes_received);
 
-    // A UDP socket that we listen on for protobuf messages from SSL Game Controller
+    // A UDP socket that we listen on for ReceiveProto messages from the network
     boost::asio::ip::udp::socket socket_;
-    // The endpoint for the sender, in this case SSL Game Controller
+    // The endpoint for the sender
     boost::asio::ip::udp::endpoint sender_endpoint_;
 
     // The maximum length of the buffer we use to receive data packets from the network
     static constexpr unsigned int max_buffer_length = 4096;
     // Acts as a buffer to store the raw received data from the network
     std::array<char, max_buffer_length> raw_received_data_;
-    // The function to call on every received packet of vision data
-    std::function<void(Referee)> handle_function;
+    // The function to call on every received packet of ReceiveProto data
+    std::function<void(ReceiveProto)> handle_function;
 };
+
+// place all templated initializations here
+template class ProtoMulticastListener<Referee>;
```

Comment:
What's the purpose of templated initializations? [From what I read here(https://en.cppreference.com/w/cpp/language/class_template) the only benefit is faster compilation?

Reply:
Its to do with the fact that we have  tempting in the header, I ran into this issue with the RobotCommunicator as well

link: https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl

Reply:
added a comment with that link

Reply:
Ah thanks for the link :+1:. I'd strongly prefer to either put the function implementations in the header or do a `.tpp` solution like gareth suggested below, because this solution relies on

> Tell your C++ compiler which instantiations to make while it is compiling your template function’s .cpp file.

Which IMO is a huge violation of the generality of templates and why we use them in the first place. I don't think it's easy for people to keep doing that in the future as things get added, they'll just get strange linker errors and be confused.

Reply:
I did the tpp solution